boards/shared/Src/Io/Io_SharedFreqOnlyPwmInput.c

```diff
@@ -46,12 +46,10 @@ struct FreqOnlyPwmInput *Io_SharedFreqOnlyPwmInput_Create(
     HAL_TIM_ActiveChannel tim_active_channel)
 {
     assert(htim != NULL);
-    static struct FreqOnlyPwmInput pwm_inputs[MAX_NUM_OF_FREQ_ONLY_PWM_INPUTS];
-    static size_t                  alloc_index = 0;
 
-    assert(alloc_index < MAX_NUM_OF_FREQ_ONLY_PWM_INPUTS);
-
-    struct FreqOnlyPwmInput *const pwm_input = &pwm_inputs[alloc_index++];
+    struct FreqOnlyPwmInput *const pwm_input =
+        (struct FreqOnlyPwmInput *)malloc(sizeof(struct FreqOnlyPwmInput));
```

Comment:
Honestly in C you don't really need to cast malloc https://stackoverflow.com/questions/605845/do-i-cast-the-result-of-malloc/605858#605858. Up to you.